### PR TITLE
Optimize InheritanceMarginGlyph construction

### PR DESCRIPTION
**Pull request by sharwell**
This change reduces the amount of work required to render Inheritance Margin glyphs which users did not directly interact with.

* Move Inheritance Margin context menu to a separate control (lazily initialized)
* Initialize Inheritance Margin glyphs in code; eliminate BAML parsing
* Optimize one-way bindings for Inheritance Margin glyphs

### CODE REVIEW

**Comment by Cosifne** [comment]
: )  Actually I am working on a similar change.
I think it would be better to let @CyrusNajmabadi to test the change first

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginContextMenu.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            using var context = _operationExecutor.BeginExecute(
+                title: EditorFeaturesResources.Navigating,
+                defaultDescription: string.Format(ServicesVSResources.Navigate_to_0, viewModel.DisplayContent),
+                allowCancellation: true,
+                showProgress: false);
+
+            var cancellationToken = context.UserCancellationToken;
+            var rehydrated = await viewModel.DefinitionItem.TryRehydrateAsync(cancellationToken).ConfigureAwait(false);
+            if (rehydrated == null)
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginContextMenu.xaml.cs:72]: aside @Cosifne this is an unfortunate way to do things.  it means that navigation could take a large amount of time just to realize the symbol just to get its' location.  For source symbols we could greatly speed this up by just store the DocIdSpan and navigating immediately to that.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginGlyph.xaml.cs [hunk 2]
```diff
@@ -50,53 +50,85 @@ internal partial class InheritanceMarginGlyph
...
 
-        private void InheritanceMargin_OnClick(object sender, RoutedEventArgs e)
+        protected override void OnMouseEnter(MouseEventArgs e)
         {
-            if (this.ContextMenu != null)
-            {
-                this.ContextMenu.IsOpen = true;
-                e.Handled = true;
-            }
+            LazyInitialize();
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginGlyph.xaml.cs:91]: 😢 I really don't like these random calls to LazyInitialize(), but I don't know a clean way to lazily-construct a context menu.

Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginGlyph.xaml.cs:91]: Am trying this alternative
https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/how-to-handle-the-contextmenuopening-event?view=netframeworkdesktop-4.8

Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMarginGlyph.xaml.cs]: ➡️ OK this is way better now.

**Review by Cosifne — APPROVED** [review]

**Pull request merged**